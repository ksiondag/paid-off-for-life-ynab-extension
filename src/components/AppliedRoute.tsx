import * as React from "react";
import { Route } from "react-router-dom";

import { AppliedRouteObj } from "../interfaces";

export default function AppliedRoute({ component: C, appProps, ...rest }: AppliedRouteObj) {
    return (
        <Route {...rest} render={props => <C {...props} {...appProps} />} />
    );
}