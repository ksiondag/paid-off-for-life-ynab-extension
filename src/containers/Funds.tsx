import * as React from "react";

import { Table, ButtonToolbar, Button, FormControl } from "react-bootstrap";

import { RouterAppProps } from "../interfaces";

import AddFund from "./funds/AddFund";

import "./Funds.css";
import {poflAccounts} from "../api/ynab";

interface Fund {
    id: number;
}

export default function Funds(props: React.PropsWithChildren<RouterAppProps>) {
    const [funds, setFunds] = React.useState([]);
    const [addFund, setAddFund] = React.useState(false);

    React.useEffect(() => {
        loadFunds();
    }, []);

    const loadFunds = async () => {
        setFunds(await poflAccounts());
        console.log(`Load funds!`);
    };

    return (
        <div className="Funds">
            <Table>
                <thead>
                    <tr>
                        <th>Fund Name</th>
                        <th>Balance</th>
                        {/* <th>
                            <ButtonToolbar>
                                <Button onClick={() => setAddFund(true)}>Add</Button>
                            </ButtonToolbar>
                        </th> */}
                    </tr>
                    {/* {addFund
                        ?
                        <AddFund setAddFund={setAddFund} saveFund={saveFund} />
                        :
                        <></>
                    } */}
                    {
                        funds.map(({ id, name, balance }) => {
                            return <tr key={id}>
                                <td>{name}</td>
                                <td>{balance}</td>
                                {/* <td>
                                    <ButtonToolbar>
                                        <Button bsStyle="danger" onClick={() => deleteFund(id)}>Delete</Button>
                                    </ButtonToolbar>
                                </td> */}
                            </tr>
                        })
                    }
                </thead>
            </Table>
        </div>
    );
}