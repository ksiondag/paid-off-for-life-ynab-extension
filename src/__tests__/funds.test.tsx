import * as React from "react";
import { MemoryRouter } from "react-router-dom";
import { render, wait, fireEvent, screen } from "@testing-library/react";

import fundsApi from "../api/funds";
import AppliedRoute from "../components/AppliedRoute";
import Funds from "../containers/Funds";

jest.mock("../api/funds");
jest.mock("../api/token");

const mockFundsApi = fundsApi as jest.Mocked<typeof fundsApi>;

afterEach(() => {
    jest.resetAllMocks()
})


test(`funds render as a table of names and balances`, async () => {
    const appProps = {
        isAuthenticated: false,
        userHasAuthenticated: (): null => null,
    };

    mockFundsApi.get.mockResolvedValue({
        success: true,
        funds: [
            { id: 11, name: `Woo`, balance: 0, balanceDate: `2020-02-08`, userId: 1 },
            { id: 12, name: `Another`, balance: 0, balanceDate: `2020-02-08`, userId: 1 },
            { id: 13, name: `Oh really?`, balance: 250, balanceDate: `2020-02-08`, userId: 1 },
            { id: 14, name: `Nick`, balance: 100000, balanceDate: `2020-02-09`, userId: 1 },
            { id: 15, name: `type a thing`, balance: 0, balanceDate: `2020-02-09`, userId: 1 },
            { id: 16, name: `yas`, balance: 250, balanceDate: `2020-02-09`, userId: 1 }
        ]
    });

    const component = render(
        <MemoryRouter>
            <AppliedRoute component={Funds} appProps={appProps} />
        </MemoryRouter>
    );
    await wait();

    expect(component).toMatchSnapshot();
});

test(`add fund`, async () => {
    const appProps = {
        isAuthenticated: false,
        userHasAuthenticated: (): null => null,
    };

    mockFundsApi.get.mockResolvedValue({
        success: true,
        funds: [
            { id: 11, name: `Woo`, balance: 0, balanceDate: `2020-02-08`, userId: 1 },
        ]
    });

    const component = render(
        <MemoryRouter>
            <AppliedRoute component={Funds} appProps={appProps} />
        </MemoryRouter>
    );
    await wait();

    expect(component).toMatchSnapshot();

    fireEvent.click(screen.getByText(`Add`));
    await wait();
    expect(component).toMatchSnapshot();
});