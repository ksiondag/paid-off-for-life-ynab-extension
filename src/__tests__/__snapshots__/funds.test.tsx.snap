// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add fund 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Funds"
      >
        <table
          class="table"
        >
          <thead>
            <tr>
              <th>
                Fund Name
              </th>
              <th>
                Balance
              </th>
              <th>
                <div
                  class="btn-toolbar"
                  role="toolbar"
                >
                  <button
                    class="btn btn-default"
                    type="button"
                  >
                    Add
                  </button>
                </div>
              </th>
            </tr>
            <tr>
              <td>
                Woo
              </td>
              <td>
                0
              </td>
              <td>
                <div
                  class="btn-toolbar"
                  role="toolbar"
                >
                  <button
                    class="btn btn-danger"
                    type="button"
                  >
                    Delete
                  </button>
                </div>
              </td>
            </tr>
          </thead>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Funds"
    >
      <table
        class="table"
      >
        <thead>
          <tr>
            <th>
              Fund Name
            </th>
            <th>
              Balance
            </th>
            <th>
              <div
                class="btn-toolbar"
                role="toolbar"
              >
                <button
                  class="btn btn-default"
                  type="button"
                >
                  Add
                </button>
              </div>
            </th>
          </tr>
          <tr>
            <td>
              Woo
            </td>
            <td>
              0
            </td>
            <td>
              <div
                class="btn-toolbar"
                role="toolbar"
              >
                <button
                  class="btn btn-danger"
                  type="button"
                >
                  Delete
                </button>
              </div>
            </td>
          </tr>
        </thead>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`add fund 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Funds"
      >
        <table
          class="table"
        >
          <thead>
            <tr>
              <th>
                Fund Name
              </th>
              <th>
                Balance
              </th>
              <th>
                <div
                  class="btn-toolbar"
                  role="toolbar"
                >
                  <button
                    class="btn btn-default"
                    type="button"
                  >
                    Add
                  </button>
                </div>
              </th>
            </tr>
            <tr>
              <td>
                <form>
                  <input
                    class="form-control"
                    placeholder="Fund name"
                    type="text"
                    value=""
                  />
                </form>
              </td>
              <td>
                <form>
                  <input
                    class="form-control"
                    placeholder="0.00"
                    type="text"
                    value="0"
                  />
                </form>
              </td>
              <td>
                <div
                  class="btn-toolbar"
                  role="toolbar"
                >
                  <button
                    class="btn btn-default"
                    type="button"
                  >
                    Save
                  </button>
                  <button
                    class="btn btn-default"
                    type="button"
                  >
                    Cancel
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td>
                Woo
              </td>
              <td>
                0
              </td>
              <td>
                <div
                  class="btn-toolbar"
                  role="toolbar"
                >
                  <button
                    class="btn btn-danger"
                    type="button"
                  >
                    Delete
                  </button>
                </div>
              </td>
            </tr>
          </thead>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Funds"
    >
      <table
        class="table"
      >
        <thead>
          <tr>
            <th>
              Fund Name
            </th>
            <th>
              Balance
            </th>
            <th>
              <div
                class="btn-toolbar"
                role="toolbar"
              >
                <button
                  class="btn btn-default"
                  type="button"
                >
                  Add
                </button>
              </div>
            </th>
          </tr>
          <tr>
            <td>
              <form>
                <input
                  class="form-control"
                  placeholder="Fund name"
                  type="text"
                  value=""
                />
              </form>
            </td>
            <td>
              <form>
                <input
                  class="form-control"
                  placeholder="0.00"
                  type="text"
                  value="0"
                />
              </form>
            </td>
            <td>
              <div
                class="btn-toolbar"
                role="toolbar"
              >
                <button
                  class="btn btn-default"
                  type="button"
                >
                  Save
                </button>
                <button
                  class="btn btn-default"
                  type="button"
                >
                  Cancel
                </button>
              </div>
            </td>
          </tr>
          <tr>
            <td>
              Woo
            </td>
            <td>
              0
            </td>
            <td>
              <div
                class="btn-toolbar"
                role="toolbar"
              >
                <button
                  class="btn btn-danger"
                  type="button"
                >
                  Delete
                </button>
              </div>
            </td>
          </tr>
        </thead>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`funds render as a table of names and balances 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Funds"
      >
        <table
          class="table"
        >
          <thead>
            <tr>
              <th>
                Fund Name
              </th>
              <th>
                Balance
              </th>
              <th>
                <div
                  class="btn-toolbar"
                  role="toolbar"
                >
                  <button
                    class="btn btn-default"
                    type="button"
                  >
                    Add
                  </button>
                </div>
              </th>
            </tr>
            <tr>
              <td>
                Woo
              </td>
              <td>
                0
              </td>
              <td>
                <div
                  class="btn-toolbar"
                  role="toolbar"
                >
                  <button
                    class="btn btn-danger"
                    type="button"
                  >
                    Delete
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td>
                Another
              </td>
              <td>
                0
              </td>
              <td>
                <div
                  class="btn-toolbar"
                  role="toolbar"
                >
                  <button
                    class="btn btn-danger"
                    type="button"
                  >
                    Delete
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td>
                Oh really?
              </td>
              <td>
                250
              </td>
              <td>
                <div
                  class="btn-toolbar"
                  role="toolbar"
                >
                  <button
                    class="btn btn-danger"
                    type="button"
                  >
                    Delete
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td>
                Nick
              </td>
              <td>
                100000
              </td>
              <td>
                <div
                  class="btn-toolbar"
                  role="toolbar"
                >
                  <button
                    class="btn btn-danger"
                    type="button"
                  >
                    Delete
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td>
                type a thing
              </td>
              <td>
                0
              </td>
              <td>
                <div
                  class="btn-toolbar"
                  role="toolbar"
                >
                  <button
                    class="btn btn-danger"
                    type="button"
                  >
                    Delete
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td>
                yas
              </td>
              <td>
                250
              </td>
              <td>
                <div
                  class="btn-toolbar"
                  role="toolbar"
                >
                  <button
                    class="btn btn-danger"
                    type="button"
                  >
                    Delete
                  </button>
                </div>
              </td>
            </tr>
          </thead>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Funds"
    >
      <table
        class="table"
      >
        <thead>
          <tr>
            <th>
              Fund Name
            </th>
            <th>
              Balance
            </th>
            <th>
              <div
                class="btn-toolbar"
                role="toolbar"
              >
                <button
                  class="btn btn-default"
                  type="button"
                >
                  Add
                </button>
              </div>
            </th>
          </tr>
          <tr>
            <td>
              Woo
            </td>
            <td>
              0
            </td>
            <td>
              <div
                class="btn-toolbar"
                role="toolbar"
              >
                <button
                  class="btn btn-danger"
                  type="button"
                >
                  Delete
                </button>
              </div>
            </td>
          </tr>
          <tr>
            <td>
              Another
            </td>
            <td>
              0
            </td>
            <td>
              <div
                class="btn-toolbar"
                role="toolbar"
              >
                <button
                  class="btn btn-danger"
                  type="button"
                >
                  Delete
                </button>
              </div>
            </td>
          </tr>
          <tr>
            <td>
              Oh really?
            </td>
            <td>
              250
            </td>
            <td>
              <div
                class="btn-toolbar"
                role="toolbar"
              >
                <button
                  class="btn btn-danger"
                  type="button"
                >
                  Delete
                </button>
              </div>
            </td>
          </tr>
          <tr>
            <td>
              Nick
            </td>
            <td>
              100000
            </td>
            <td>
              <div
                class="btn-toolbar"
                role="toolbar"
              >
                <button
                  class="btn btn-danger"
                  type="button"
                >
                  Delete
                </button>
              </div>
            </td>
          </tr>
          <tr>
            <td>
              type a thing
            </td>
            <td>
              0
            </td>
            <td>
              <div
                class="btn-toolbar"
                role="toolbar"
              >
                <button
                  class="btn btn-danger"
                  type="button"
                >
                  Delete
                </button>
              </div>
            </td>
          </tr>
          <tr>
            <td>
              yas
            </td>
            <td>
              250
            </td>
            <td>
              <div
                class="btn-toolbar"
                role="toolbar"
              >
                <button
                  class="btn btn-danger"
                  type="button"
                >
                  Delete
                </button>
              </div>
            </td>
          </tr>
        </thead>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
